### PR TITLE
IGAPP-452: Create prettier config

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

I also already added the configs for Intellij because it is not possible to configure prettier to exactly match our current formatting rules, due to their [option philosophy](https://prettier.io/docs/en/option-philosophy.html). So we have to reformat all files with prettier. (e.g. it is not possible to have a space between function name and braces -> render () becomes render().) 

Now our linting rules only care about code-quality rules, and prettier takes care of formatting rules (overriding every es-lint rule about formatting). This avoids conflicts of prettier and eslint. Prettier runs autmatically on save in Intellij and vs code. 

If a prettier rule is violated it is not visible in the IDE, we can add this feature by adding eslint-plugin-prettier package, which added prettier as a linting rule, but this is not recommended and therefore i did not add it, but wanted to mention it. So it would make sense to add prettier checks to ci in the future, when all files are fixed.

### CODE REVIEW

File: .circleci/src/commands/lint.yml [hunk 1]
```diff
@@ -6,5 +6,8 @@ steps:
   - run:
       name: Linting check
       command: yarn workspace << parameters.directory >> run lint:ci
+  # - run: TODO
```
Comment (klinzo) [inline .circleci/src/commands/lint.yml:9]: Should be enabled before merging together with a PR applying all new prettier rules.

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "semi": false,
+  "trailingComma": "none",
+  "singleQuote": true,
+  "arrowParens": "avoid",
+  "printWidth": 120,
+  "jsxBracketSameLine": false,
```
Comment (klinzo) [inline .prettierrc:7]: I think this should be true.

File: README.md [hunk 3]
```diff
@@ -57,6 +57,7 @@ We suggest **[IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/)** as IDE.
 If you are using a different IDE like Visual Studio Code ([./docs/vscode.md](./docs/vscode.md)) the steps might differ.
 
 * Import this project (VCS > Get from Version Control).
+* Install Prettier Plugin (inteillij should recogzie everything and format code with prettier, if not got to Settings -> Languages and Framworks | Javascript | Prettier and select the installed prettier package)
```
Comment (klinzo) [inline README.md:60]:
```suggestion
* Install Prettier Plugin (IntelliJ should recognize everything and format code with prettier, if not go to Settings > Languages and Frameworks > Javascript > Prettier and select the installed prettier package)
```

File: docs/conventions.md [hunk 4]
```diff
@@ -8,6 +8,7 @@ We follow the [airbnb style](https://github.com/airbnb/javascript/tree/master/re
 
 We mainly follow the [standard style](https://standardjs.com/rules) for JavaScript, although there may be smaller deviations.
 Run the linter (`yarn lint`) to show problems. Our eslint config can be found [here](https://github.com/Integreat/eslint-config-integreat).
+We use prettier to format code. Run (`yarn workspace web (or native) prettier --check .`) to show formatting problems.
```
Comment (klinzo) [inline docs/conventions.md:11]:
```suggestion
We use prettier to format code. Run (`yarn workspace <workspace> prettier --check .`) to show formatting problems.
```

File: .circleci/src/commands/lint.yml (hunk 1 above)
Comment (SteffiStuffel) [reply .circleci/src/commands/lint.yml:9]: done in second PR

File: .prettierrc (hunk 2 above)
Comment (SteffiStuffel) [reply .prettierrc]: changed it.

**Review by Taggotty — APPROVED** [review]

**Review by klinzo — APPROVED** [review]

**Pull request merged**